internal/class.h

```diff
@@ -37,9 +37,9 @@ struct rb_classext_struct {
     struct st_table *iv_index_tbl;
     struct st_table *iv_tbl;
 #if SIZEOF_SERIAL_T == SIZEOF_VALUE /* otherwise m_tbl is in struct RClass */
-    struct rb_id_table *m_tbl;
+    struct rb_managed_id_table *m_tbl;
```

Comment:
(trivial) renaming `m_tbl` will helpful to detect the code which doesn't respect the type (using cast)  and macro.